pyaqueduct/client/client.py

```diff
@@ -480,13 +471,85 @@ def execute_extension_action(
                 error.errors if error.errors else "Unknown error occurred in the remote operation."
             ) from error
 
-        result = ExtensionExecutionResultData.from_dict(
-            extension_result["executeExtension"]  # pylint: disable=unsubscriptable-object
-        )
+        result = ExtensionExecutionResultData.from_dict(extension_result["executeExtension"]) # pylint: disable=unsubscriptable-object
         logging.info(
             "Executed a %s / %s extension action with result code %d",
             extension,
             action,
             result.returnCode,
         )
         return result
+
+    def get_task(self, task_id: str) -> TaskData:
+        """Get details for a submitted taks
+
+        Args:
+            task_id: Task identifier
+        """
+        task_result = self.fetch_response(
```

Comment:
Do you re-request data from backend again for each task? Please reuse results of your generic query